Microsoft.Azure.CosmosRepository/src/ChangeFeed/IContainerChangeFeedProcessor.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.CosmosRepository.ChangeFeed
+{
+    internal interface IContainerChangeFeedProcessor
+    {
+        Task StartAsync();
```

Comment:
I'm curious if a `CancellationToken` should flow through start/stop

Reply:
It hits a bit of a dead-end, when starting the change feed processor I have had to add this but not sure if gives many benefits what do you think?

```csharp
 public async Task StartAsync(CancellationToken cancellationToken)
{
    Container itemContainer = await _containerService.GetContainerAsync(ItemTypes);
    Container leaseContainer = await _leaseContainerProvider.GetLeaseContainerAsync();

    _processor = itemContainer.GetChangeFeedProcessorBuilder<JObject>("cosmos-repository-pattern-processor",
            (changes, token) => OnChanges(changes, token, itemContainer.Id))
        .WithLeaseContainer(leaseContainer)
        .WithInstanceName(_changeFeedOptions.InstanceName)
        .WithErrorNotification((token, exception) => OnError(exception, itemContainer.Id))
        .Build();

    cancellationToken.Register(() => StopAsync(cancellationToken));

    _logger.LogInformation("Starting change feed processor for container {ContainerName}", itemContainer.Id);

    await _processor.StartAsync();

    _logger.LogInformation("Successfully started change feed processor for container {ContainerName}",
        itemContainer.Id);
}
```

Reply:
I'm not certain either. I'd just hate to have a bug reported at a later time where someone couldn't get cancellation to work as expected.

Reply:
Yes I agree, I will put a sample together to show the cancellation as well as testing it, will make sure this happens before a release.